bindings/serialize.h

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SERIALIZE
+#define SERIALIZE
+
+#include <memory>
+#include <map>
+#include "v8-profiler.h"
+
+std::vector<char> serializeHeapProfile(std::unique_ptr<v8::AllocationProfile>
+  profile, int64_t intervalBytes, int64_t startTimeNanos);
+
+std::vector<char> serializeTimeProfile(std::unique_ptr<v8::CpuProfile>,
+  int64_t samplingIntervalMicros, int64_t startTimeNanos);
+
+class ProtoMessage {
```

Comment:
I don't think it's a proto message type since you use it for primitive types. Proto message is an aggregate type which contains other messages and fields. This is more like ProtoField. Take a look at the existing C++ proto implementation to align on terminology.

Reply:
Done.